examples/cryptocurrency-advanced/docker/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM ubuntu:16.04
+
+ENV ROCKSDB_LIB_DIR=/usr/lib/x86_64-linux-gnu
+ENV SNAPPY_LIB_DIR=/usr/lib/x86_64-linux-gnu
+
+RUN apt-get update \
+  && apt-get install -y curl git gcc cpp \
+  libssl-dev pkg-config libsodium-dev libsnappy-dev librocksdb-dev
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable
+
+RUN curl -sL https://deb.nodesource.com/setup_8.x | bash \
+  && apt-get install -y nodejs
+
+RUN mv /root/.cargo/bin/* /usr/bin
```

Comment:
I'd suggest to combine with the `cargo ...` commands below. Also, I wonder if Rust is still needed after the installation of the cryptocurrency binary - it may make sense to remove it to reduce the Docker image bloat, or use a separate builder container, [like here](https://github.com/emk/rust-musl-builder/blob/master/examples/using-diesel/Dockerfile).